### DISCUSSION TITLE
복수 서브스크립트 (코드 17-4) 질문!

### DISCUSSION THREAD

**Discussion by hyunjung-ann** [swift-in-gangnam/swift-grammar#27] [Q&A] [answered]
복수 서브스크립트 코드 17-4 (324p) 공부하다가 filter 관련 이해 안가는 부분이 있어 질문드려요! 

**1.**
` subscript(name:String) -> Int? {
        get {
            return self.students.filter{ $0.name == name}.first?.number`

-> 해석하면, `students`의 각 항목 이름마다 전달인자의 이름과 똑같을시, 첫번째 요소의 number(=인덱스)를 반환한다 뜻일까요? 
    first.number 요부분이 헷갈리네요 ㅠㅠ 

**2.**
`subscript(name:String,number:Int) -> Student? {
        return self.students.filter {$0.name == name && $0.number == number }.first }` 
-> 위 코드랑 같은 맥락으로 `students`의 각 항목 이름과 번호가 전달인자 값과 똑같다면 첫번째 요소 반환한다 뜻일까요??

**Comment by Chuncheonian** [comment, marked as answer]
저도 이부분이 어려워서 많이 봤었네요ㅠㅠ 확실히 Optional이 어렵긴해요ㅠ

1.
![첫번째](https://user-images.githubusercontent.com/29334434/126147849-4ef0717f-0c56-460a-829b-9629b2fe3d44.jpg)

2. 네 맞아요! name과 number를 전달받으면, students 배열 안의 원소 중에 같은 name과 number를 가진 구조체를 반환합니다.

**Reply by hyunjung-ann** [reply]
오... 그림까지!! 👍🏻 이렇게 설명해주시니까 이해하기가 한결 더 쉬웠어요!!  저 코드안에 저번 파트 때 배웠던게 다 들어있었네요 ㅋㅋㅋ 옵셔널 체이닝,필터 ㅠㅠ 덕분에 해결했습니다 ! 감사합니다 ㅎㅎㅎ
